src/API/index.js

```diff
@@ -72,3 +72,14 @@ export const fetchLaunchTemplates = async (sourceID, region) => {
   } = await axios.get(provisioningUrl(`sources/${sourceID}/launch_templates?region=${region}`));
   return data;
 };
+
+export const checkPermissions = async (provider, sourceID, region) => {
+  switch (provider) {
+    case AWS_PROVIDER: {
+      const { data } = await axios.get(provisioningUrl(`sources/${sourceID}/validate_permissions?region=${region}`));
+      return data;
+    }
+    default:
+      return [];
```

Comment:
`react-query` function needs to be a promise, I double-checked, and returning anything in an `async` block, returns a promise with the status `resolved`, so it should work as expected :)